pyomo/contrib/parmest/parmest.py

```diff
@@ -613,19 +1036,169 @@ def _Q_opt(
                     if len(vals) > 0:
                         var_values.append(vals)
                 var_values = pd.DataFrame(var_values)
-                if calc_cov:
-                    return objval, thetavals, var_values, cov
-                else:
-                    return objval, thetavals, var_values
 
-            if calc_cov:
-                return objval, thetavals, cov
-            else:
-                return objval, thetavals
+                return objval, thetavals, var_values
+
+            return objval, thetavals
 
         else:
             raise RuntimeError("Unknown solver in Q_Opt=" + solver)
 
+    def _cov_at_theta(self, method, solver, cov_n, step):
+        """
+        Covariance matrix calculation using all scenarios in the data
+
+        Argument:
+            method: string ``method`` object specified by the user (e.g., kaug)
+            solver: string ``solver`` object specified by the user (e.g., ipopt)
+            cov_n: integer, number of datapoints specified by the user which is used in the objective function
+            step: float used to perturb the parameters
+
+        Returns:
+            cov: pd.DataFrame, covariance matrix of the estimated parameters
+        """
+        # Number of data points considered
+        n = cov_n
+
+        # Extract number of fitted parameters
+        l = len(self.estimated_theta)
+
+        # Assumption: Objective value is sum of squared errors
+        sse = self.objective_value
+
+        '''Calculate covariance assuming experimental observation errors are
+        independent and follow a Gaussian distribution with constant variance.
+
+        The formula used in parmest was verified against equations (7-5-15) and
+        (7-5-16) in "Nonlinear Parameter Estimation", Y. Bard, 1974.
+
+        This formula is also applicable if the objective is scaled by a constant;
+        the constant cancels out. (was scaled by 1/n because it computes an
+        expected value.)
+        '''
+        # check if the supplied method is supported
+        try:
+            cov_method = CovMethodLib(method)
+        except:
```

Comment:
Should this catch a specific type of exception from Enum? As written, this is catching all possible exceptions.